tests/e2e/performance/test_pvc_multi_snapshot_performance.py

```diff
@@ -0,0 +1,156 @@
+            f'Total capacity size is : {ceph_capacity} GiB, '
+            f'Going to use {need_capacity} GiB, '
+            f'With {num_of_snaps} Snapshots to {pvc_size} GiB PVC. '
+            f'File size to be written is : {file_size} '
+            f'with the name of {file_name}'
+        )
+
+        snap_yaml = constants.CSI_RBD_SNAPSHOT_YAML
+        if self.interface == constants.CEPHFILESYSTEM:
+            snap_yaml = constants.CSI_CEPHFS_SNAPSHOT_YAML
+
+        for test_num in range(1, (num_of_snaps + 1)):
+
+            test_results = {
+                'test_num': test_num,
+                'create': {'time': None, 'speed': None},
+            }
+
+            # Going to run only write IO to fill the PVC for the snapshot.
+            log.info(
+                f'Running IO on pod {self.pod_obj.name} - Test num. {test_num}'
+            )
+            self.pod_obj.fillup_fs(size=file_size, fio_filename=file_name)
+
+            # Wait for fio to finish
+            fio_result = self.pod_obj.get_fio_results()
+            err_count = fio_result.get('jobs')[0].get('error')
+            assert err_count == 0, (
+                f"IO error on pod {self.pod_obj.name}. "
+                f"FIO result: {fio_result}"
+            )
+            log.info('IO on the PVC Finished')
+
+            # Take a snapshot of the PVC and measure the time of creation.
+            snap_name = self.pvc_obj.name.replace(
+                'pvc-test', f'snapshot-test{test_num}'
+            )
+            log.info(f'Taking snapshot of the PVC {snap_name}')
+            snap_obj = pvc.create_pvc_snapshot(
+                self.pvc_obj.name,
+                snap_yaml,
+                snap_name,
+                helpers.default_volumesnapshotclass(self.interface).name,
+            )
+            snap_obj.ocp.wait_for_resource(
+                condition='true', resource_name=snap_obj.name,
+                column=constants.STATUS_READYTOUSE, timeout=60
+            )
+
+            snap_con_name = snap_obj.ocp.get(
+                resource_name=snap_name,
+                out_yaml_format=True
+            )["status"]["boundVolumeSnapshotContentName"]
+            log.info(f'Snap content name in test-{test_num} is {snap_con_name}')
+
+            test_results['create']['time'] = helpers.measure_snapshot_creation_time(
+                self.interface, snap_obj.name, snap_con_name
+            )
+            test_results['create']['speed'] = int(
+                filesize * constants.GB2MB / test_results['create']['time']
+            )
+            log.info(f'Test {test_num} results:')
+            log.info(f'Snapshot creation time is : {test_results["create"]["time"]} sec.')
+            log.info(f'Snapshot speed is : {test_results["create"]["speed"]} MB/sec')
+            create_times.append(test_results)
+
+        logging.info(f"Snapshot created results are:  {create_times}")
+
+        # TODO: push all results to elasticsearch server
```

Comment:
should we maintain an issue to track this?

Reply:
will open one.